src/components/NavBarMenuButtons.tsx

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import {
+  Dialog,
+  DialogTrigger,
+  Menu,
+  MenuItem,
+  MenuTrigger,
+  Popover,
+  PopoverProps,
+} from "react-aria-components";
+import styled from "styled-components";
+import { colors } from "../../src/theme";
+import { NavBarButton, NavBarButtonProps } from "./NavBarButton";
+
+export const NavBarMenuItem = styled(MenuItem)``;
+
+export const PopoverContainer = styled.div`
+  padding: 1.6rem;
+`;
+
+const NavBarPopover = styled(Popover)`
+  margin-top: -1rem;
+  border-top: 0.4rem solid ${colors.palette.darkGreen};
+  box-shadow: 0 0.4rem 0.4rem 0 #00000033;
+  background: #fff;
+
+  ${NavBarMenuItem} {
+    font-size: 1.4rem;
+    min-height: 4rem;
+    padding: 0 1.6rem;
+    cursor: pointer;
+    display: flex;
+    align-items: center;
+
+    &:hover {
+      background: ${colors.palette.neutralLighter};
+    }
+
+    &:active {
+      font-weight: bold;
+    }
+
+    &:not(:last-child) {
+      border-bottom: 0.1rem solid ${colors.palette.neutralBright};
+    }
+  }
+`;
+
+export type NavBarItemProps = React.PropsWithChildren<{
+  label?: string;
+  icon?: string | React.ReactNode;
+  buttonProps?: NavBarButtonProps;
+  popoverProps?: PopoverProps;
+  className?: string;
+}> & (
+  | { label: string; icon?: React.ReactNode; "aria-label"?: string }
```

Comment:
Extra `|` at the beginning of the line here?

Reply:
This entire type can probably be DRYed up. Or at least label and icon can be removed from the stuff inside React.PropsWithChildren. Also, I don't think `icon: string` is allowed in the end based on the types in the OR.

Reply:
I guess prettier likes to put `|` at the beginning of multi line types like this 🤷. Good call about DRYing it up, using the NavBarButtonProps as a base let me delete a lot of type code.